File: iOS/JobLoyal/UIKit/App/User/Profile/UserProfileTableViewController/Model/UserProfileModel.swift
//
//  UserProfileModel.swift
//  JobLoyal
//
//  Created by Sina khanjani on 4/5/1400 AP.
//

import Foundation

// MARK: - UserProfileModel
struct UserProfileModel: Codable, Hashable {
    let id, name, family: String?
    let gender: Bool?
    let email, address, phoneNumber, birthday: String?
    let region, createdAt, updatedAt: String?
    let credit: Double?

    enum CodingKeys: String, CodingKey {
        case id, name, family, gender, email, address
        case phoneNumber = "phone_number"
        case birthday, region, createdAt, updatedAt, credit
    }
}

struct UserInfoMenuModel: Codable, Hashable {
    let name, family, phoneNumber: String?
    
    init(userProfileModel: UserProfileModel) {
        name = userProfileModel.name
        family = userProfileModel.family
        phoneNumber = userProfileModel.phoneNumber
    }
}

File: android/app/src/main/java/com/jobloyal/user/model/profile/UserProfileModel.kt
package com.jobloyal.user.model.profile

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class UserProfileModel(
    val address: String?,
    val birthday: String?,
    val createdAt: String?,
    val credit: Double?,
    val email: String?,
    val family: String?,
    val gender: Boolean?,
    val id: String?,
    val name: String?,
    val phone_number: String?,
    val region: String?,
    val updatedAt: String?
) : Parcelable